MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs

```diff
@@ -179,37 +195,87 @@ internal static Version GetFFmpegVersion(string output)
             else
             {
-                // Create a reduced version string and lookup key from dictionary
-                var reducedVersion = GetLibrariesVersionString(output);
+                if (!TryGetFFmpegLibraryVersions(output, out string versionString, out IReadOnlyDictionary<string, double> versionMap))
+                {
+                    _logger.LogError("No ffmpeg library versions found");
+
+                    return null;
+                }
+
+                // First try to lookup the full version string
+                if (_ffmpegVersionMap.TryGetValue(versionString, out Version version))
+                {
+                    return version;
+                }
+
+                // Then try to test for minimum library versions
+                return TestMinimumFFmpegLibraryVersions(versionMap);
+            }
+        }
+
+        private Version TestMinimumFFmpegLibraryVersions(IReadOnlyDictionary<string, double> versionMap)
+        {
+            var allVersionsValidated = true;
 
-                // Try to lookup the string and return Key, otherwise if not found returns null
-                return _ffmpegVersionMap.TryGetValue(reducedVersion, out Version version) ? version : null;
+            foreach (var minimumVersion in _ffmpegMinimumLibraryVersions)
+            {
+                if (versionMap.TryGetValue(minimumVersion.Key, out var foundVersion))
+                {
+                    if (foundVersion >= minimumVersion.Value)
+                    {
+                        _logger.LogInformation("Found {Library} version {FoundVersion} ({MinimumVersion})", minimumVersion.Key, foundVersion, minimumVersion.Value);
+                    }
+                    else
+                    {
+                        _logger.LogWarning("Found {Library} version {FoundVersion} lower than recommended version {MinimumVersion}", minimumVersion.Key, foundVersion, minimumVersion.Value);
+                        allVersionsValidated = false;
+                    }
+                }
+                else
+                {
+                    _logger.LogError("{Library} version not found", minimumVersion.Key);
+                    allVersionsValidated = false;
+                }
             }
+
+            return allVersionsValidated ? MinVersion : null;
         }
 
         /// <summary>
         /// Grabs the library names and major.minor version numbers from the 'ffmpeg -version' output
-        /// and condenses them on to one line.  Output format is "name1=major.minor,name2=major.minor,etc."
         /// </summary>
         /// <param name="output"></param>
+        /// <param name="versionString"></param>
+        /// <param name="versionMap"></param>
         /// <returns></returns>
-        private static string GetLibrariesVersionString(string output)
+        private static bool TryGetFFmpegLibraryVersions(string output, out string versionString, out IReadOnlyDictionary<string, double> versionMap)
```

Comment:
Why does this return both the version string AND a version dictionary?

Reply:
The version string is used to test against the old version map (_ffmpegVersionMap) created by ffmpeg -version | some regex. If that test fails the dictionary is used to test against the minimum versions (_ffmpegMinimumLibraryVersions)

Reply:
Maybe it would make sense to remove the old version map and just rely on the minimum library versions as the fallback, but that wasn't my intent. Let me know if you want it changed.